src/modules/UI/scenes/Plugins/EdgeProvider.js

```diff
@@ -163,18 +163,55 @@ export class EdgeProvider extends Bridgeable {
     if (walletId && currencyCode) {
       this.selectedWallet = this._state.core.account.currencyWallets[walletId]
       if (this.selectedWallet == null) throw new Error(`Missing wallet for walletId`)
+      const chainCode = this.selectedWallet.currencyInfo.currencyCode
+      const tokenCode = currencyCode
       const { pluginId } = this.selectedWallet.currencyInfo
       const tokenId = getTokenId(this._state.core.account, pluginId, currencyCode)
 
-      // If allowedCurrencyCodes is an array of EdgeTokenIdExtended objects
+      let returnCurrencyCode = ''
+
       if (allowedCurrencyCodes != null && allowedCurrencyCodes.length > 0 && allowedCurrencyCodes.every(code => typeof code === 'object')) {
+        // If allowedCurrencyCodes is an array of EdgeTokenIdExtended objects
         return {
           pluginId,
           tokenId,
           currencyCode
         }
       }
-      return unfixCurrencyCode(this._plugin.fixCurrencyCodes, pluginId, tokenId) ?? currencyCode
+      if (chainCode !== tokenCode) {
+        // User selected a token
+        if (chainCode === 'ETH') {
+          // Special case for ETH. Caller can specify a token code alone and it can be matched against ETH tokens
+          // Check if the tokenCode also exists as a parent chain currencyCode. ie. "MATIC"
+          const codeLookup = makeCurrencyCodeTable(this._state.core.account)
+          const edgeToken = codeLookup(tokenCode)
+          if (edgeToken.find(t => t.tokenId == null)) {
+            // Found a mainnet chain with the same currencyCode as this token. Return the full currency code. ie. "ETH-MATIC"
+            returnCurrencyCode = `ETH-${tokenCode}`
+          } else {
+            // See if tokenCode matches one of the allowedCurrencyCodes
+            returnCurrencyCode = (allowedCurrencyCodes ?? []).find(m => m === tokenCode)
+          }
+
+          // If no match found. Check if a full currency code was used.
+          if (returnCurrencyCode == null) {
+            returnCurrencyCode = (allowedCurrencyCodes ?? []).find(m => m === `ETH-${tokenCode}`)
+          }
+          if (returnCurrencyCode == null) {
+            throw new Error(`Internal error. Tokencode ${tokenCode} selected but not in allowedCurrencyCodes`)
+          }
+        } else {
+          // Tokens for all other chains must be specified using a full currency code. ie. "ETH-USDC"
+          returnCurrencyCode = `${chainCode}-${tokenCode}`
+        }
+      } else {
```

Comment:
For readability, these blocks could be flattened and each conditional returned immediately.

Reply:
Agreed there's only a little improvement to be had but will do the cleanup in a new PR